src/gui/instrument/LfoGraph.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * EnvelopeGraph.cpp - Displays LFO graphs
+ *
+ * Copyright (c) 2004-2014 Tobias Doerffel <tobydox/at/users.sourceforge.net>
+ * Copyright (c) 2024-     Michael Gregorius
+ * 
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include <QMouseEvent>
+#include <QPainter>
+
+#include "LfoGraph.h"
+
+#include "EnvelopeAndLfoParameters.h"
+#include "Oscillator.h"
+
+#include "gui_templates.h"
+
+namespace lmms
+{
+
+extern const float SECS_PER_LFO_OSCILLATION;
+
+namespace gui
+{
+
+LfoGraph::LfoGraph(QWidget* parent) :
+	QWidget(parent),
+	ModelView(nullptr, this),
+	m_params(nullptr)
+{
+	setFixedSize(m_lfoGraph.size());
+}
+
+void LfoGraph::modelChanged()
+{
+	m_params = castModel<EnvelopeAndLfoParameters>();
+}
+
+void LfoGraph::mousePressEvent(QMouseEvent* me)
+{
+	if (me->button() == Qt::LeftButton)
+	{
+		toggleAmountModel();
+	}
+}
+
+void LfoGraph::paintEvent(QPaintEvent*)
```

Comment:
A number of local variables in the method don't follow the naming conventions.

Reply:
I think these should be addressed when the `LfoGraph` is adjusted to be able to render itself at any size. This will bring lots of restructuring anyway.